DataFormats/Common/interface/RefCoreWithIndex.h

```diff
@@ -33,6 +33,12 @@ namespace edm {
                      bool transient,
                      unsigned int elementIndex);
 
+    RefCoreWithIndex(ProductID const& id, void const* prodPtr, unsigned int index)
```

Comment:
Please add a comment stating that `transient` will be set to `false` for this constructor.